examples/compute_head_direction.py

```diff
@@ -0,0 +1,397 @@
+
+# %%
+# Compute the "forward" vector
+# ----------------------------
+# We can also compute head direction using the
+# :func:`compute_forward_vector()<movement.kinematics.compute_forward_vector>`
+# function, which takes a different approach to the one we used above:
+# it accepts a pair of bilaterally symmetric keypoints and
+# computes the vector that's perpendicular to the line connecting them.
+#
+# Here we will use the two ears to find the head direction vector.
+# We may prefer this method if we expect the snout detection to be
+# unreliable (e.g., because it's often occluded in a top-down camera view).
+
+from movement.kinematics import (
+    compute_forward_vector,
+    compute_forward_vector_angle,
+)
+
+forward_vector = compute_forward_vector(
+    position,
+    left_keypoint="left_ear",
+    right_keypoint="right_ear",
+    camera_view="top_down",
+)
+print(forward_vector)
+
+# %%
+# .. admonition:: Why do we need to specify the camera view?
+#   :class: note
+#
+#   You can think about it in this way: in order to uniquely determine which
+#   way is forward for an animal, we need to know the orientation of the other
+#   two body axes: left-right and up-down. The left-right axis is specified
+#   by the left and right keypoints passed to the function, while we use the
+#   ``camera_view`` parameter to determine the upward direction (see image).
+#   The default view is ``"top_down"``, but it can also be ``"bottom_up"``.
+#   Other camera views are not supported at the moment.
+#
+#   .. image:: ../_static/Forward-Vector.png
+#     :width: 600
+#     :alt: Schematic showing forward vector in top-down and bottom-up views
+
+
+# %%
+# You can use ``compute_forward_vector`` to compute the direction
+# of any body segment, as long as bilaterally symmetric keypoints are
+# available. For example, you could estimate body direction given the
+# the two hips or shoulders.
+#
+# Specifically for the head direction vector, you may also use the alias
+# :func:`compute_head_direction_vector()\
+# <movement.kinematics.compute_head_direction_vector>`,
+# which makes the intent of the function clearer.
+
+# %%
+# Compute head direction angle
+# ----------------------------
+# We could compute the head direction angle from the forward vector as
+# we did with the head-to-snout vector, i.e., by converting the vector to
+# polar coordinates and extracting the ``phi`` coordinate. However, it's
+# more convenient to use the :func:`compute_forward_vector_angle()\
+# <movement.kinematics.compute_forward_vector_angle>` function, which
+# directly returns the angle between the forward vector and the positive
+# x-axis.
+
+forward_vector_angle = compute_forward_vector_angle(
+    position,
+    left_keypoint="left_ear",
+    right_keypoint="right_ear",
+    # Optional parameters:
+    reference_vector=(1, 0),  # positive x-axis
+    camera_view="top_down",
+    in_radians=True,  # set to False for degrees
+)
+print(forward_vector_angle)
```

Comment:
Here instead of printing you may want to consider:
- plotting the forward vector and the positive x-axis for a single specific frame, when it is at an angle that is visually easy to check (maybe when theta is close to +(pi/4) rad = +45deg), and
- printing that single angle value

That may make the point more clear.

Reply:
I agree it will make the point clearer, but I'm worried that the example is already to long. With every additional "validation" plot we add, we may increase clarity, but at the cost of length and also at the cost of adding plotting code that may detract from the main points we are trying to make.

We have to make that trade-off decision for every single plot in this example, and here I've decided against it.
I will re-evaluate once I've responded to all other comments and I check back on how the whole thing flows (and how long it is).